lib/vanagon/component/rules.rb

```diff
@@ -48,17 +48,26 @@ def initialize(component, project, platform)
       #
       # @return [Array<Makefile::Rule>]
       def rules
-        list = [
-          component_rule,
-          unpack_rule,
-          patch_rule,
-          configure_rule,
-          build_rule,
-          check_rule,
-          install_rule,
-          clean_rule,
-          clobber_rule,
-        ]
+        if component.install_only
+          list = [
```

Comment:
Using something less generic than `list` seems appropriate. Maybe something `list_of_rules`?

Minor: I would also change the pattern to initialize & override rather than if/else:

```ruby
list_of_rules = [
  default_rule1,
  default_rule2,
  ...
]

if component.install_only
  list_of_rules  = [
    override_rule1,
    override_rule2,
    ...
  ]
end
```

At this point, it doesn't matter all that much but it could increase maintainability/refactorability in the future.

Reply:
Good call, thanks @e-gris !

Reply:
Any time I see an 'else', it triggers scrutiny. There are times where they make perfect sense but they are surprisingly few.